### ISSUE TITLE
realtek-poe daemon reload interrupts power to PoE sinks

### ISSUE THREAD

**Issue by mrnuke** [Hurricos/realtek-poe#5]
## Summarry

It has been reported in [this OpenWRT forum post](https://forum.openwrt.org/t/bringing-support-for-realtek-poe-to-mainline-openwrt/127777/25?u=mrnuke) that realtek-poe behavior is really annoying.  PoE devices lose power whenever realtek-poe is started or restarted.

This is undesireable for two reasons:
1. It's annoying
2. Crashes in realtek-poe will shut down PoE devices.

## Technical details

The following two lines in poe_initial_setup() cause the power to PoE ports to be cut off when realtek-poe is started:

https://github.com/Hurricos/realtek-poe/blob/db485c85b2c6c997049b81c44a3c0d6d57d1728c/src/main.c#L712-L713

## Course of action
On my engenius-ews2910p with 22.10 firmware, the MCU does not care about the state of the ports. It will happily accept a "set global power budget" command and apply it dynamically.
- Is this behavior reliable on other FW versions or MCUs?

If so, then we could simply remove the calls in question, and everything should work,

**Comment by Hurricos** [comment]
From https://github.com/Hurricos/realtek-poe/pull/6#issuecomment-1166650607:

> The [gpio-hogdefinition](https://github.com/openwrt/openwrt/blob/326e109f24731d751171f7a21bd8f5a0a9123613/target/linux/realtek/dts-5.10/rtl8380_zyxel_gs1900.dtsi#L43-L55) looks fine to me. I wonder if u-boot is forcing the GPIO low.

I can watch the line. I'm worried that OpenWrt is setting it low. Let me attempt to invoke a reboot from the OEM FW.

**Comment by mrnuke** [comment]
There's a difference between a system reboot and a daemon restart. This issue is focused on shutting down and restarting realtek poe. For example:

    /etc/init.d/poe restart

On a system reboot we really don't have that much control. If the bootloader decides to reset the PoE MCU, or disable the poe-enable line, there is little we can do to prevent that. Only in the event that the bootloader, and gpio-restart hardware do not touch the PSEs, do we have the power to not cut power.

**Comment by Hurricos** [comment]
No, I meant that I was worried that something in the kernel shutdown routine might be setting it low ("unhogging"). It was unfounded.

There *is* an interesting U-boot command on these ZyXEL switches called `cst pinSet/pinGet`:

```txt
RTL838x# cst pinGet
Usage: cst pinGet <pinNum> 
 pinNum: 0~36
```

It is (probably) connected to the [RTL8231 on the SFP side of the board](https://wikidevi.wi-cat.ru/images/c/c1/GS1900-24HP-main-pcb.JPG) (as `cst pinSet` never changes an LED status light).

Behavior:
- pins 0-2 are read-only (do not respond to pinSet)
- pin 3 is default '1', but 1s after accepting a write, it bounces back:
```txt
RTL838x# cst pinSet 3 0
pin3:   0

RTL838x# cst pinGet 3  
pin3:   0

RTL838x# cst pinGet 3  
pin3:   1
```

- Running `cst pinSet 5 0` reboots the switch, (pin 3 on `gpio1`, maybe?)
- Appears that the rest of the pins are read-only. This includes "pin 13".

I can open the device and trace these GPIOs a bit to see what U-boot is actually doing here.

I'm not actually with the switch -- doing this remotely now -- but I plan to follow up on this one:

> Let me attempt to invoke a reboot from the OEM FW.

... once I can actually get shell access to the PoE sink (so I can verify it's rebooted when I reboot the ZyXEL switch, as I have no clue how to check link state under ZyXEL's OS).

**Comment by mrnuke** [comment]
Can we move the reboot discussion into its own issue?

**Comment by mrnuke** [comment]
With #6 I will be removing the call to `poe_cmd_global_power_budget{}`. That only leaves `poe_cmd_global_port_enable(0);` which I think should just be removed as well.

**Comment by walterav1984** [comment]
This seems to be fixed for the CI22 realtek-poe package release, as in no interruption for a currently PoE powered device while changing /etc/config/poe and doing a `/etc/init.d/poe reload` or `ubus call poe reload` and powering on a other PoE device. Have to test extensive "disabling" and "enabling" in /etc/config/poe yet though.

**Comment by mrnuke** [comment]
`/etc/init.d/poe reload` Does not stop the daemon, but `/etc/init.d/poe restart` does.

**Comment by Hurricos** [comment]
Issue fixed by b1e21d6c1e6bebd07321ce9045f8865f2c338a80.

**Comment by walterav1984** [comment]
Besides the non-interupted PoE `ubus call poe reload` or `/etc/init.d/poe reload` I can also confirm that even `/etc/init.d/poe restart` won't interupt PoE power anymore on already powered on PoE devices with package CI29.
